prime-angular-frontend/src/app/modules/enrolment/pages/obo-sites-page/obo-sites-page.component.ts

```diff
@@ -211,13 +217,40 @@ export class OboSitesPageComponent extends BaseEnrolmentProfilePage implements O
         this.setOboSites();
       });
     } else {
-      //for renewal, pull the enrolment from enrolment service instead
+
       this.enrolment = this.enrolmentService.enrolment;
-      this.enrolmentFormStateService.patchOboSitesForm(this.enrolment.oboSites);
+      if (this.lastUpdatedDatetime.value == null) {
+        //for renewal, pull the enrolment from enrolment service instead
+        this.enrolmentFormStateService.patchOboSitesForm(this.filterOboSitesByCareSetting(this.enrolment.oboSites));
+      } else {
+        let oboSites = this.enrolmentFormStateService.json.oboSites;
+        this.enrolmentFormStateService.patchOboSitesForm(oboSites);
+      }
+
       this.setOboSites();
     }
   }
 
+  //filter obo site by the care setting in the form - remove any site not in the care settings
+  private filterOboSitesByCareSetting(oboSites: OboSite[]): OboSite[] {
+    for (var i = 0; i < oboSites.length;) {
```

Comment:
Could the same be achieved by going through the array from last element to first?

Reply:
Again, I don't see advantage one way rather than other. It is just the way I implemented.

Reply:
It is updated.